src/ray/object_manager/object_buffer_pool.h

```diff
@@ -0,0 +1,186 @@
+#ifndef RAY_OBJECT_MANAGER_OBJECT_BUFFER_POOL_H
+#define RAY_OBJECT_MANAGER_OBJECT_BUFFER_POOL_H
+
+#include <list>
+#include <memory>
+#include <mutex>
+#include <vector>
+
+#include <boost/asio.hpp>
+#include <boost/asio/error.hpp>
+#include <boost/bind.hpp>
+
+#include "plasma/client.h"
+#include "plasma/events.h"
+#include "plasma/plasma.h"
+
+#include "ray/id.h"
+#include "ray/status.h"
+
+namespace ray {
+
+/// \class ObjectBufferPool Exposes chunks of object buffers for use by the ObjectManager.
+class ObjectBufferPool {
+ public:
+  /// Information needed about each object chunk.
+  /// This is the structure returned whenever an object chunk is
+  /// retrieved.
+  struct ChunkInfo {
```

Comment:
Would it work if we just returned a `uint8_t *` to the data buffer instead of defining this other struct?

Reply:
We still need buffer length, but I did removed unnecessary fields in this struct.